scene/main/node.cpp

```diff
@@ -2121,7 +2121,7 @@ Node *Node::_duplicate(int p_flags, Map<const Node *, Node *> *r_duplimap) const
 			for (int i = 0; i < N->get()->get_child_count(); ++i) {
 
 				// Skip nodes not really belonging to the instanced hierarchy; they'll be processed normally later
-				if (get_child(i)->data.owner != this)
+				if (N->get()->get_child(i)->data.owner != this)
 					continue;
```

Comment:
`i` is intended to index children of `N->get()`. It goes up to `N->get()->get_child_count() - 1`, but you are using it to index children of `this`. I don't think this is correct.

Could you explain the logic behind this fix?

Reply:
It's the other way round, isn't it? It was indexing `this` and now it indexes `N->get()`.

Reply:
Indeed. I swear I double and triple-checked my observation! :P